### ISSUE TITLE
no-dynamic-delete rule does not accept readonly variables and strings

### ISSUE THREAD

**Issue by Martin-Wegner** [palantir/tslint#3648]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.6.2
- __Running TSLint via__: CLI and VSCode

#### TypeScript code being linted

```ts
private static readonly STATIC_KEY: string = 'foo';
private readonly KEY: string = 'bar';

delete fromControl.errors['foobar'];
delete fromControl.errors[MyComponent.STATIC_KEY];
delete fromControl.errors[this.key];
```

with `tslint.json` configuration:

```json
"no-dynamic-delete": true
```

#### Actual behavior
tslint complains about all 3 lines with ` Do not delete dynamically computed property keys`.

#### Expected behavior
All 3 keys are non computed properties.
You should allow readonly variables and simple strings (this can be forbidden with the `no-string-literal` rule) or change the rule description and error message and make it more clear, please.

**Comment by JoshuaKGoldberg** [comment]
Agreed that there should be a way to enable this behavior. How about an opt-in rule option to allow it?

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
